crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,705 @@
+        &self,
+        instructions: String,
+        blobs: Blobs,
+        message: Message,
+        is_wallet_transaction: bool,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionToReview> {
+        let network_id = self.profile_state_holder.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+        let transaction_manifest =
+            TransactionManifest::new(instructions, network_id, blobs)?;
+
+        // Get the transaction preview
+        let transaction_preview = self
+            .get_transaction_preview(
+                gateway_client,
+                transaction_manifest.clone(),
+                network_id,
+                message,
+                nonce,
+                notary_public_key,
+            )
+            .await?;
+        let engine_toolkit_receipt = transaction_preview
+            .radix_engine_toolkit_receipt
+            .ok_or(CommonError::FailedToExtractTransactionReceiptBytes)?;
+
+        // Analyze the manifest
+        let execution_summary = transaction_manifest
+            .execution_summary_with_engine_toolkit_receipt(
+                engine_toolkit_receipt,
+            )?;
+
+        // Transactions created outside of the Wallet are not allowed to use reserved instructions
+        if !is_wallet_transaction
+            && !execution_summary.reserved_instructions.is_empty()
+        {
+            return Err(
+                CommonError::ReservedInstructionsNotAllowedInManifest {
+                    reserved_instructions: execution_summary
+                        .reserved_instructions
+                        .iter()
+                        .map(|i| i.to_string())
+                        .collect(),
+                },
+            );
+        }
+
+        Ok(TransactionToReview {
+            transaction_manifest,
+            execution_summary,
+        })
+    }
+
+    async fn get_transaction_preview(
+        &self,
+        gateway_client: GatewayClient,
+        manifest: TransactionManifest,
+        network_id: NetworkID,
+        message: Message,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionPreviewResponse> {
+        let signer_public_keys =
+            self.extract_transaction_signers(manifest.summary()).await?;
+        let epoch = gateway_client.current_epoch().await?;
+        let header = TransactionHeader::new(
+            network_id,
+            epoch,
+            Epoch::from(epoch.0 + 10),
+            nonce,
+            notary_public_key,
+            signer_public_keys.is_empty(),
+            0,
+        );
+        let intent = TransactionIntent::new(header, manifest, message)?;
+        let request = TransactionPreviewRequest::new(
+            intent,
+            signer_public_keys,
+            TransactionPreviewRequestFlags::new(true, false, false),
```

Comment:
ah ok, hmm can you please make newtypes for these three flags. That should have been done before IMO. since `true, false, false` is hard to read. Better with `IsFoo(true), IsBar(false), IsBuz(false)`.

There is a macro for this! see:

```rust
decl_bool_type!(IsCloudProfileSyncEnabled, true);
decl_bool_type!(IsDeveloperModeEnabled, false);
decl_bool_type!(IsAdvancedLockEnabled, false);
```
the bool literal is the default value